colossalai/tensor/_ops/addmm.py

```diff
@@ -70,22 +70,19 @@ def colo_addmm(types, args, kwargs, pg):
 
     # Add communication logic before and after linear call.
     ret_tensor = None
-    if not mat2.has_spec():    # No Model Parallel Applied
-        assert not input_tensor.has_spec(), 'Invalid input spec for native addmm op'
+    if mat2.spec.is_gathered():    # No Model Parallel Applied
```

Comment:
So spec must exist now?

Reply:
`has_spec()` actually checks whether the colo tensor has parallel actions. And now, all colo tensors have `.spec`. If TP is used, `has_spec()` must be true. If not, no matter whether `has_spec()` is true is OK.

Reply:
I see